js/main.js

```diff
@@ -44,97 +43,188 @@ const GUESTS_AMOUNT = [
-            rooms: getRandomData(ROOMS_AMOUNT),
-            guests: getRandomData(GUESTS_AMOUNT),
-            checkin: getRandomData(CHECKINS),
-            checkout: getRandomData(CHECKOUTS),
-            features: getRandomData(FEATURES),
-            description: ` `,
-            photos: getRandomData(PHOTOS_URLS)
-          }
-        }
-    );
+    array.push({
+      author: {
+        avatar: `img/avatars/user0${i + 1}.png`
+      },
+      location: {
+        x: getRandomInt(Coordinates.X.MIN, Coordinates.X.MAX),
+        y: getRandomInt(Coordinates.Y.MIN, Coordinates.Y.MAX)
+      },
+      offer: {
+        title: getRandomArrElement(TITLES),
+        address: `${getRandomInt(Coordinates.X.MIN, Coordinates.X.MAX)}, ${getRandomInt(Coordinates.Y.MIN, Coordinates.Y.MAX)}`,
+        price: getRandomInt(Price.MIN, Price.MAX),
+        type: getRandomArrElement(Object.keys(HOUSE_TYPES)),
+        rooms: getRandomArrElement(ROOMS_AMOUNT),
+        guests: getRandomArrElement(GUESTS_AMOUNT),
+        checkin: getRandomArrElement(CHECKINS),
+        checkout: getRandomArrElement(CHECKOUTS),
+        features: getRandomValuesArr(Object.keys(FEATURES_CLASS_MAP)),
+        description: getRandomArrElement(DESCRIPTIONS),
+        photos: getRandomValuesArr(PHOTO_URLS)
+      }
+    });
   }
   return array;
 };
 
-const createPin = (array) => {
+const createPin = (obj) => {
   const pinElement = mapPinTemplate.cloneNode(true);
-  pinElement.style.left = `${array.location.x - 25}px`;
-  pinElement.style.top = `${array.location.y - 35}px`;
-  pinElement.querySelector(`img`).src = array.author.avatar;
-  pinElement.querySelector(`img`).alt = array.offer.title;
+  pinElement.style.left = `${obj.location.x}px`;
+  pinElement.style.top = `${obj.location.y}px`;
+  pinElement.querySelector(`img`).src = obj.author.avatar;
+  pinElement.querySelector(`img`).alt = obj.offer.title;
 
   return pinElement;
 };
 
-const createNodeFragment = (pin) => {
+const createCard = (dataObject) => {
+  const cardElement = mapCardTemplate.cloneNode(true);
+  if (dataObject.offer.title.length) {
+    cardElement.querySelector(`.popup__title`).classList.remove(`hidden`);
+    cardElement.querySelector(`.popup__title`).textContent = dataObject.offer.title;
+  }
+  if (dataObject.offer.address.length) {
+    cardElement.querySelector(`.popup__text--address`).classList.remove(`hidden`);
+    cardElement.querySelector(`.popup__text--address`).textContent = dataObject.offer.address;
+  }
+  if (dataObject.offer.price.length) {
+    cardElement.querySelector(`.popup__text--price`).classList.remove(`hidden`);
+    cardElement.querySelector(`.popup__text--price`).textContent = `${dataObject.offer.price} ₽/ночь`;
+  }
+  if (dataObject.offer.price.length) {
+    cardElement.querySelector(`.popup__text--price`).classList.remove(`hidden`);
+    cardElement.querySelector(`.popup__text--price`).textContent = `${dataObject.offer.price} ₽/ночь`;
+  }
```

Comment:
Тут повтор и баг. У числового литерала нет свойства `length`, потому условие не срабатывает и цена не отображается
```suggestion
  if (dataObject.offer.price) {
    cardElement.querySelector(`.popup__text--price`).classList.remove(`hidden`);
    cardElement.querySelector(`.popup__text--price`).textContent = `${dataObject.offer.price} ₽/ночь`;
  }
```

![image](https://user-images.githubusercontent.com/30810655/94526688-8a82fc00-023e-11eb-9a0d-a57bc8184d48.png)

-----

К слову, если тебе когда-нибудь понадобится локализовать числа или даты, то для этого есть специальный глобальный конструктор `Intl`. Я им пользуюсь время от времени для локализации цены товаров, весьма удобно. [Примеры использования](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat#Example_Using_locales)